### ISSUE TITLE
Attribute for size of classes in vision/torchvision/datasets/folder.py

### ISSUE THREAD

**Issue by deepaksharma36** [pytorch/vision#431]
It would be great if you can provide one method inside ImageFolder class or Dataset class which can provide number of samples for each loaded class/label.

Though we can do this outside ImageFolder class, but as we always use ImageFolder for loading image dataset hence a class_size attribute will quickly tell us how many samples of each class/label are loaded.   
   from collections import Counter   
Counting num of samples outside ImageFolder class:
    image_datasets['train'] = datasets.ImageFolder(os.path.join(DATA_DIR, 'train'))
    class_counts['train'] = dict(Counter(sample_tup[1] for sample_tup in image_datasets['train''].imgs))

**Comment by fmassa** [comment]
While I think there might be value in having such a functionality, for consistency we might want to add a similar functionality to other datasets in torchvision as well.

Also, as you have pointed out, this is fairly easy to do using standard python, so I'm unsure about what would be the best thing to do here.

What are the use-cases of having the number of samples for each class? I can think of a few:
- sampling the elements of a dataset with a probability which is a function of the class occurrences, using `WeightedRandomSampler`
- for debugging

Any other use-case?

**Comment by deepaksharma36** [comment]
Most of the time while dealing with unbalanced datasets we go for a weighed loss function, where weights are decided based on number of samples for the class. We assign higher weights to the rare class so that model don't get biased toward the class with higher samples. 

class_weights = [1-(float(class_counts['train'][class_id])/(dataset_sizes['train'])) for class_id in range(NUM_CLASSES)]  
criterion = nn.CrossEntropyLoss(weight=torch.FloatTensor(class_weights).cuda())

**Comment by ProGamerGov** [comment]
I've just come across a need for this feature to calculate CrossEntropyLoss loss weights.
